enum.c

```diff
@@ -156,18 +162,28 @@ enum_grep(VALUE obj, VALUE pat)
 }
 
 /*
- *  call-seq:
- *     enum.grep_v(pattern)                  -> array
- *     enum.grep_v(pattern) { |obj| block }  -> array
+ * call-seq:
+ *   grep_v(pattern) -> new_array
+ *   grep_v(pattern) {|element| ... } -> new_array
+ *
+ * Returns a new \Array of objects either selected by the given +pattern+
+ *  or returned by the block.
  *
- *  Inverted version of Enumerable#grep.
- *  Returns an array of every element in <i>enum</i> for which
- *  not <code>Pattern === element</code>.
+ * With no block given, returns a new \Array containing each element
+ * for which <tt>pattern === element</tt> is +false+:
  *
- *     (1..10).grep_v 2..5   #=> [1, 6, 7, 8, 9, 10]
- *     res =(1..10).grep_v(2..5) { |v| v * 2 }
- *     res                    #=> [2, 12, 14, 16, 18, 20]
+ *   a = Float.constants
+ *   a.grep_v(/MIN|MAX/)              # => [:ROUNDS, :RADIX, :MANT_DIG, :DIG, :EPSILON, :INFINITY, :NAN]
+ *   (1..10).grep_v(3..8)             # => [1, 2, 9, 10]
+ *   ['a', 'b', 0, 1].grep_v(Integer) # => ["a", "b"]
  *
+ * With a block given,
+ * calls the block with each element, returns a new \Array containing each
+ * object returned by the block:
+ *
+ *   a = Float.constants
```

Comment:
I think the original block example using a range is better, because it is easier to fully comprehend.